src/odemis/gui/cont/streams.py

```diff
@@ -122,6 +126,45 @@ def __init__(self, stream_bar, stream, tab_data_model, show_panel=True, view=Non
         self._lbl_exc_peak = None
         self._lbl_em_peak = None
 
+        # Display metadata for integration time, ebeam voltage, probe current and
+        # emission/excitation wavelength
+        if show_md:
+            if isinstance(stream.raw[0], tuple):
+                # Show metadata for first element of raw in case of AR acquisitions or
+                # stitched images
+                md = stream.raw[0][0].metadata
+            else:
+                md = stream.raw[0].metadata
+
+            # Add exposure time and dwell time to integration time
+            if model.MD_EXP_TIME in md.keys() and model.MD_DWELL_TIME in md.keys():
+                self.add_metadata("Integration Time", md[model.MD_EXP_TIME] + md[model.MD_DWELL_TIME], 's')
+            elif model.MD_EXP_TIME in md.keys():
+                self.add_metadata("Integration Time", md[model.MD_EXP_TIME], 's')
+            elif model.MD_DWELL_TIME in md.keys():
+                self.add_metadata("Integration Time", md[model.MD_DWELL_TIME], 's')
+
+            if model.MD_EBEAM_VOLTAGE in md.keys():
+                self.add_metadata(model.MD_EBEAM_VOLTAGE, md[model.MD_EBEAM_VOLTAGE], 'V')
+
+            if model.MD_EBEAM_CURRENT in md.keys():
+                self.add_metadata(model.MD_EBEAM_CURRENT, md[model.MD_EBEAM_CURRENT], 'A')
+
+            if model.MD_OUT_WL in md.keys():
+                em_range = md[model.MD_OUT_WL]
+                if isinstance(em_range, basestring):
+                    unit = None
+                else:
+                    unit = "m"
+                self.stream.emission = model.VigilantAttribute(em_range, unit=unit,
```

Comment:
No. You _cannot_ create new attribute on another object (Python allows you, but really, you should not).
Two possibilities:
 * directly create the right (Text) widget to show the emission/excitation.
 * put back the code in the stream constructor.

If the stream actually needs a .emission or .excitation VA (for other parts of the GUI to work properly), you have to do the second alternative. My feeling is that anyway, you could just do that (ie, revert this change), and everything will be fine.